tests/test_dataframe_checks.py

```diff
@@ -170,3 +170,17 @@ def test_contains_inf(df_na):
     df_na.fillna(np.inf, inplace=True)
     with pytest.raises(ValueError):
         assert _check_contains_inf(df_na, ["Age", "Marks"])
+
+
+def test_check_X_raises_error_on_duplicated_column_names():
+    df = pd.DataFrame(
+        {
+            "col1": [1, 2, 3],
+            "col2": ["a", "b", "c"],
+            "col3": pd.date_range("2023-01-01", periods=3),
+        }
+    )
+    df.columns = ["same", "unique", "same"]
+
+    with pytest.raises(ValueError):
```

Comment:
nitpick: could we add that it tests the error message as well? And example on how to do so here:

https://github.com/feature-engine/feature_engine/blob/3343305a01d1acfeff846b65d33a5686c6e8c84f/tests/test_encoding/test_woe/test_woe_encoder.py#L189-L197